plugins/frontend/backstage-plugin-cloudsmith/src/plugin.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2022 Larder Software Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createApiFactory,
+  createPlugin,
+  FetchApi,
+  fetchApiRef,
+  DiscoveryApi,
+  discoveryApiRef,
+} from '@backstage/core-plugin-api';
+import { createCardExtension } from '@backstage/plugin-home';
+
+import { CloudsmithClient, cloudsmithApiRef } from './api';
+
+export const cloudsmithPlugin = createPlugin({
+  id: 'cloudsmith',
+  apis: [
+    createApiFactory({
+      api: cloudsmithApiRef,
+      deps: { fetchApi: fetchApiRef, discoveryApi: discoveryApiRef },
```

Comment:
You might be missing
`      factory: ({ discoveryApi, fetchApi }) => new CloudsmithClient({ discoveryApi, fetchApi})
`

Reply:
Isnt that on the next line?

Reply:
I didn't see it, sorry 😟